win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1035,8 +1103,8 @@ static BOOL CURSORICON_GetCursorDataFromANI(
             }
             else
             {
-                pFrameData->cx = pHeader->width;
-                pFrameData->cy = pHeader->height;
+                if (!pFrameData->cx) pFrameData->cx = pHeader->width;
```

Comment:
It was pointed out,
QUESTION:
Why are you only assigning if cx and Co. is 0? 

You can have a image that’s not actually a image no?

Reply:
> Why are you only assigning if cx and Co. is 0?

@DarkFire01 Because `pHeader` is received from ANI header, and `pHeader->width` and `pHeader->height` contain the actual size of the cursor (1 : 1), while `pCurData->cx` and `pCurData->cy` are properly initialized values regarding to the system metrics/user definitions. I initialize these properly as well, see my changes above, at lines 1078 and 1079. :wink:  `pDirEntry` condition is never hit btw, since `pHearer` metric values are always valid (non-zero).
If we will always write `pHeader`'s metrtics values, they will always overwrite the correct metrics-defined values. So I made an additional condition for them: use them only when the previous values are incorrect/failed to be set properly. Since we don't need to always use a real cursor size (for example when it's too big), we use the system metrics values instead.
It has nothing to do with normal sized cursors, but I found a non-standard sized cursor theme, which has 256x256 actual size of the cursors. It has the visible difference before and after this fix.
:white_check_mark: I tested it on Windows XP SP3 and on Windows Server 2003 SP2 as well too: these cursors also use a system metrics defined size there too, so this repeats (is more correct regarding to) the Windows behaviour. :slightly_smiling_face: